### PR TITLE
[Audio]: added conformer U-Net model for SE

### PR DESCRIPTION
**Pull request by nasretdinovr**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

This PR adds the conformer U-Net architecture for Speech Enhancement and unit tests for it.

**Collection**: audio

# Changelog

- added
nemo/collections/audio/parts/submodules/conformer_unet.py
examples/audio/conf/streaming_predictive_conformer_unet.yaml
examples/audio/conf/predictive_conformer_unet.yaml
- modified 
tests/collections/audio/test_audio_models_predictive.py

# Usage

```
python examples/audio/audio_to_audio_train.py --config-path=conf --config-name=streaming_predictive_conformer_unet.yaml model.train_ds.cuts_path=/path/to/cuts_manifest.jsonl model.validation_ds.cuts_path=/path/to/cuts_manifest.jsonl

```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [-] Did you add or update any necessary documentation?
- [-] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: tests/collections/audio/test_audio_models_predictive.py [hunk 1]
```diff
@@ -566,3 +707,118 @@ def test_adaptive_rms_ebabled_fails(self, predictive_model_transformer_unet, bat
             # fail because of adaptive RMS turned on for predictive model
             with torch.no_grad():
                 _, _ = model.forward(input_signal=input_signal, input_length=input_signal_length)
+
+class TestPredictiveModelConformerUNet:
```
Comment (anteju) [inline tests/collections/audio/test_audio_models_predictive.py:829]: 👍

File: nemo/collections/audio/parts/submodules/conformer_unet.py [hunk 2]
```diff
@@ -0,0 +1,1401 @@
...
+from nemo.core.classes.exportable import Exportable
+from nemo.core.classes.mixins import AccessMixin, adapter_mixins
+from nemo.core.classes.module import NeuralModule
+from nemo.core.neural_types import AcousticEncodedRepresentation, ChannelType, LengthsType, NeuralType, SpectrogramType
+from nemo.utils import logging
+
+__all__ = ['SpectrogramConformerUNet', 'ConformerEncoderUNet']
+
+
+class ConformerEncoderUNet(NeuralModule, StreamingEncoder, Exportable, AccessMixin):
```
Comment (anteju) [inline nemo/collections/audio/parts/submodules/conformer_unet.py]: Most of this class is unchanged from `ConformerEncoder`. Since a lot of stuff is already imported from `asr`, an alternative would be for `ConformerEncoderUNet` to subclass `ConformerEncoder`.
At a first glance, in that case `ConformerEncoderUNet` would only have to apply minimal modification to `__init__` (to initialize `self.skip_*` variables and `self.skip_connector` layers), and implement `forward_internal`.
